### DISCUSSION TITLE
Kitty requires network access to theme switch?

### DISCUSSION THREAD

**Discussion by ldelossa** [kovidgoyal/kitty#4257] [Q&A] [answered]
Hello, 

I started my laptop just now with no network connection and my theme switching failed. I run the command:

```
kitty +kitten themes --reload-in=all Dark-theme
```

Where Dark-theme is my own custom theme in the config folder. 

The network access is a little alarming, can I make this not happen?

**Comment by kovidgoyal** [comment]
The themes kitten loads themes from a central repository. Sadly there is
no magical technique by which one can collect themes from a central
repository without using the network. I could certainly have it not
bother to update the themes list when only doing a reload, open a
separate issue for that so I dont forget.

**Comment by kovidgoyal** [comment, marked as answer]
Actually the facility is already there, see the --cache-age option.

**Reply by ldelossa** [reply]
Thanks a lot @kovidgoyal 

Set this to "-1" and now them switching work offline.

**Comment by IceAsteroid** [comment]
Here let me clarity that, the complete command will look like this
```
kitty +kitten themes --cache-age=-1 _themeYouWantToSwitchTo_
```
where -1 stands for no checking,
	0 stands for always checking,
	1 is the default for checking after one day,
	N is to check updates after N days.

You can find docs here: https://sw.kovidgoyal.net/kitty/kittens/themes/

I couldn't find any docs related to kittens, so, you also may need to look at the online docs above.

Also, if you are out of a kitty instance, and want to switch themes for all kitty terminals, for example, say you want to do this in a script, you can then specify

```
kitty +kitten themes --cache-age=-1 _theTheme_
kill -SIGUSR1 $(pgrep kitty)
```

To inform all the kitty instances to update the conf file, which is where the theme is changed.

This is what I found here: https://old.reddit.com/r/KittyTerminal/comments/rzpjed/easier_ways_to_reload_kitty/
